sdk-setup/src/sdk-manage

```diff
@@ -1565,13 +1565,27 @@ target_snapshot_needs_reset() {
         return 1
     fi
 
-    # TODO how to determine recent package removal?
     local most_recent_install_time=
     most_recent_install_time=$(set -o pipefail; rpm --root="$MER_TARGETS/$original" -qa \
         --queryformat '%{installtime}\n' |sort -n |tail -n1) || return
-    local snapshot_time_s=$(date --date "$snapshot_time" +%s) || return
 
-    (( most_recent_install_time > snapshot_time_s ))
+    # The rpm database does not know about the most recent package removal, so
+    # parse the zypper install log. zypp/history looks like this:
+    # 2019-02-27 16:39:18|remove |droid-hal-mydevice-tools|0.0.6-201902271629|armv7hl|user@pc|
+    # But could also look like this:
+    # 2019-02-27 16:39:18|install|remove-my-pkg|0.0.6-201902271629|armv7hl|user@pc|
+    local most_recent_removal_time=$(set -o pipefail; \
+        tac "$MER_TARGETS/$original/var/log/zypp/history" \
+        |cut -d "|" -f "1,2" \
+        |grep -m 1 "|remove" \
+        |cut -d "|" -f 1 \
+    ) || return
+    most_recent_removal_time=$(date --date "$most_recent_removal_time" +%s) || return
+
+    # Compare values in unix seconds
+    local snapshot_time_s=$(date --date "$snapshot_time" +%s) || return
+    (( most_recent_install_time > snapshot_time_s )) && return
```

Comment:
We could also test each value immediately after it is determined - if it fails, no need to determine the next one. I would consider eliminating `snapshot_time_s` by storing the seconds directly in `snapshot_time` (prepend `@` when passing to `touch --date`).

Reply:
Changed to compare and return on success early.

I have also incorporated your change regarding converting to unix seconds earlier, but imo it does not abet readability and does not really offer any speed benefits. (I really had to dig a bit to find out what `@` does in the context of `date` formatting, and it's a non-standard GNU extension.)

Reply:
Yeah, but how usable is the *nix world without GNU extensions? :) I really didn't think about speed benefits - I wanted to bring the rest of the code closer to your approach where you reuse the same variable `most_recent_removal_time` for different "data type" :)